### PR TITLE
Add example app

### PR DESCRIPTION
**Pull request by luk1337**

### CODE REVIEW

File: example/app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    id("org.jetbrains.kotlin.android")
+}
+
+apply {
+    plugin<GenerateBpPlugin>()
+}
+
+buildscript {
+    repositories {
+        mavenLocal()
```
Comment (luk1337) [inline example/app/build.gradle.kts:21]: Maybe useful to add:
```
// maven("https://raw.githubusercontent.com/lineage-next/gradle-generatebp/v1.0/.m2")
```

File: example/app/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (SebaUbuntu) [inline example/app/.gitignore]: Newline

File: example/app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    implementation("androidx.appcompat:appcompat:1.6.1")
+    implementation("com.google.android.material:material:1.9.0")
+    testImplementation("junit:junit:4.13.2")
+    androidTestImplementation("androidx.test.ext:junit:1.1.5")
+    androidTestImplementation("androidx.test.espresso:espresso-core:3.5.1")
+}
+
+configure<GenerateBpPluginExtension> {
+    targetSdk.set(android.defaultConfig.targetSdk!!)
+    availableInAOSP.set { false }
```
Comment (SebaUbuntu) [inline example/app/build.gradle.kts]: A proper example might be useful?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -15,11 +15,16 @@ jobs:
...
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'zulu'
+        java-version: 17
+
     - name: Build
       run: ./gradlew build
 
     - name: Publish
-      if: github.event_name == 'workflow_dispatch'
```
Comment (SebaUbuntu) [inline .github/workflows/build.yml:22]: Mistake?

Comment (luk1337) [reply .github/workflows/build.yml:22]: No, I want up to date .m2.

File: example/app/.gitignore (hunk 2 above)
Comment (luk1337) [reply example/app/.gitignore]: Done.

File: example/app/build.gradle.kts (hunk 3 above)
Comment (luk1337) [reply example/app/build.gradle.kts]: Done.

**Pull request merged**